lib/Dialect/RTL/Ops.cpp

```diff
@@ -427,7 +427,33 @@ void AndOp::getCanonicalizationPatterns(OwningRewritePatternList &results,
         return success();
       }
 
-      /// TODO: and(x, and(...)) -> and(x, ...) -- flatten
+      // and(x, and(...)) -> and(x, ...) -- flatten
+      for (size_t i = 0; i != size; ++i) {
+        if (!inputs[i].hasOneUse())
+          continue;
+        if (auto andOp = inputs[i].getDefiningOp<rtl::AndOp>()) {
+          auto andOpInputs = andOp.inputs();
+          SmallVector<Value, 4> newOperands;
+          newOperands.reserve(size + andOpInputs.size());
+
+          size_t j = 0;
+          for (; j != i; ++j)
+            // Push back original operands until the andOp.
+            newOperands.push_back(inputs[j]);
```

Comment:
SmallVector has an append member that would make this cleaner, something like this should work:

```
newOperands.append(inputs.begin(), inputs.begin()+i);
```
Similarly for the loop below

Reply:
Thanks! Much cleaner.